### PR TITLE
feat: Delete eco-credit balance and supply when it goes to zero

### PR DESCRIPTION
**Pull request by likhita-809**
ref: #360

### CODE REVIEW

File: x/ecocredit/server/helpers.go [hunk 1]
```diff
@@ -56,13 +55,18 @@ func getSubAndSetDecimal(store sdk.KVStore, key []byte, x *apd.Decimal) error {
 		return err
 	}
 
-	err = math.SafeSub(value, value, x)
-	if err != nil {
-		return err
+	if value.IsZero() {
```
Comment (blushi) [inline x/ecocredit/server/helpers.go:58]: shouldn't we check for zero value once we have substracted `x` from `value`?

Comment (likhita-809) [reply x/ecocredit/server/helpers.go:58]: sorry, my bad. I'll fix it

Comment (likhita-809) [reply x/ecocredit/server/helpers.go:58]: sorry, my bad. I'll fix it

**Review by blushi — COMMENTED** [review]
should we make this ready for review?

**Comment by likhita-809** [comment]
> should we make this ready for review?

**Comment by atheeshp** [comment]
looks good @likhita-809 , can you also check for zero values [here](https://github.com/regen-network/regen-ledger/blob/987fe7acea659cd491f69dcf88c5fbc675e835fe/x/ecocredit/server/msg_server.go#L119-L120).

**Comment by atheeshp** [comment]
> looks good @likhita-809 , can you also check for zero values [here](https://github.com/regen-network/regen-ledger/blob/987fe7acea659cd491f69dcf88c5fbc675e835fe/x/ecocredit/server/msg_server.go#L119-L120), while creating the batch.

instead we can revert all these changes and add a check for zero value in `setDecimal` ([here](https://github.com/regen-network/regen-ledger/blob/987fe7acea659cd491f69dcf88c5fbc675e835fe/x/ecocredit/server/helpers.go#L29-L35)) func and delete the key if matches zero. (following the same in sdk x/bank).

**Review by atheeshp — APPROVED** [review]

**Review by blushi — APPROVED** [review]

**Pull request merged**